Fix printing of `smt_get_value_commandt`

thomasspriggs opened the pull request:
<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.

If your PR fixes a bug, include the regression test(s) in the same commit as the bug fix. Else, keep commits small and orthogonal, possibly placing tests in commits of their own.
--->

This PR fixes the printing of `smt_get_value_commandt`. The get value command is supposed to be parameterised with a collection of one or more terms for which the solver should return the values. Our internal representation of the get-value command only supports a single term, which should be placed in brackets to make it into a collection containing a single term. The brackets were previously missing, which would have caused a solver error.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

thomasspriggs committed:
Add unit test for the printing of `smt_get_value_commandt`

Because this had previously been missed.

thomasspriggs committed:
Fix printing of `smt_get_value_commandt`

The get value command is supposed to be parameterised with a collection
of one or more terms for which the solver should return the values. Our
internal representation of the get-value command only supports a single
term, which should be placed in brackets to make it into a collection
containing a single term. The brackets were previously missing, which
would have caused a solver error.

TGWDB reviewed (approved):
I recognise in cbmc we only ever ask for one value at a time, maybe we should note that the standard allows multiple in the code in case someone wants to change this one day?

thomasspriggs committed:
Document limitations of `smt_get_value_commandt`

So that future maintainers can be made aware of this discrepency between
the standards and the capabilities of `smt_get_value_commandt`.
